frontend/src/context/anticipatoryActionStateSlice/index.ts

```diff
@@ -0,0 +1,367 @@
+      availableDates,
+      windowKey,
+    };
+  });
+
+  return { windowData, monitoredDistricts };
+}
+
+type AnticipatoryActionState = {
+  data: { [windowKey: string]: AnticipatoryActionData | undefined };
+  availableDates?: { [windowKey: string]: DateItem[] };
+  monitoredDistricts: string[];
+  selectedDateData: {
+    [key: string]: Pick<
+      AnticipatoryActionDataRow,
+      'district' | 'windows' | 'category' | 'phase'
+    >[];
+  };
+  selectedWindow: typeof AAWindowKeys[number] | typeof allWindowsKey;
+  categoryFilters: Record<AACategoryType, boolean>;
+  renderedDistricts: Record<
+    typeof AAWindowKeys[number],
+    {
+      [district: string]: { category: AACategoryType; phase: AAPhaseType };
+    }
+  >;
+  loading: boolean;
+  error: string | null;
+};
+
+const initialState: AnticipatoryActionState = {
+  data: {},
+  availableDates: undefined,
+  monitoredDistricts: [],
+  selectedDateData: {},
+  selectedWindow: allWindowsKey,
+  categoryFilters: {
+    Severo: true,
+    Moderado: true,
+    Leve: true,
+    na: true,
+    ny: true,
+  },
+  renderedDistricts: { 'Window 1': {}, 'Window 2': {} },
+  loading: false,
+  error: null,
+};
+
+export const loadAAData = createAsyncThunk<
+  {
+    windowData: {
+      data: AnticipatoryActionData;
+      availableDates: DateItem[];
+      windowKey: typeof AAWindowKeys[number];
+    }[];
+    monitoredDistricts: string[];
+  },
+  undefined,
+  CreateAsyncThunkTypes
+>('anticipatoryActionState/loadAAData', async () => {
+  const url = appConfig.anticipatoryActionUrl;
+
+  return new Promise<any>((resolve, reject) =>
+    Papa.parse(url, {
+      header: true,
+      download: true,
+      complete: results => resolve(transform(results.data, AACSVKeys)),
+      error: error => reject(error),
+    }),
+  );
+});
+
+interface CalculateMapRenderedDistrictsParams {
+  selectedDateData: AnticipatoryActionState['selectedDateData'];
+  categoryFilters: AnticipatoryActionState['categoryFilters'];
+  selectedWindow: AnticipatoryActionState['selectedWindow'];
+}
+
+function calculateMapRenderedDistricts({
+  selectedDateData,
+  categoryFilters,
+  selectedWindow,
+}: CalculateMapRenderedDistrictsParams) {
+  if (Object.entries(selectedDateData).length === 0) {
+    return { 'Window 1': {}, 'Window 2': {} };
+  }
+  const mapped = Object.entries(selectedDateData).map<
+    Pick<
+      AnticipatoryActionDataRow,
+      'district' | 'windows' | 'category' | 'phase'
+    >
+  >(([district, districtData]) => {
+    // eslint-disable-next-line fp/no-mutating-methods
+    const sortedData = [...districtData].sort((a, b) => {
+      const aOrder = AADataSeverityOrder(a.category, a.phase);
+      const bOrder = AADataSeverityOrder(b.category, b.phase);
+
+      if (aOrder > bOrder) {
+        return -1;
+      }
+      if (aOrder < bOrder) {
+        return 1;
+      }
+      return 0;
+    });
+
+    const filtered = sortedData.filter(x => categoryFilters[x.category]);
```

Comment:
This is most often NA I think but we can come back to this later.
This process also does not take into account the fact that once SET is attained it stays available. Or am I missing something? Is there not a way to reuse what's calculated for dataForRows in the HomeTable?

Reply:
Solved in #1170